src/ansible_compat/runtime.py

```diff
@@ -50,6 +50,7 @@ def __init__(
         self,
         project_dir: Optional[str] = None,
         isolated: bool = False,
+        version: Optional[str] = None,
```

Comment:
The docstring says that it's a minimal version. Just version may mean "install/upgrade this version if missing". But if you improve the naming, it may become clear what it is:
```suggestion
        min_required_version: Optional[str] = None,
```